scripts/macos-gen-dmg.py

```diff
@@ -26,6 +26,12 @@ def main(args):
         nargs='?',
     )
 
+    parser.add_argument('universal_dmg',
+        help='If specified, creates a universal dmg (x64/arm64)',
+        type=Path,
+        nargs='?',
+    )
+
     args = parser.parse_args(args[1:])
 
     with temp_install(args.build_dir) as install_path:
```

Comment:
I think this should be refactored a bit. The body of the `with` block should be move into a function taking `install_path` as an argument. Then you can do something like this:

```python
if args.installation is None:
    with temp_install(args.build_dir) as install_path:
        package_dmg(install_path)
else:
    package_dmg(args.installation)
```

This way we don't do a useless `temp_install()` in case we have a premade installation.